public/jwpltx.js

```diff
@@ -37,10 +10,83 @@ window.jwpltx = window.jwpltx || {};
     oosv: '5',
     sdk: '0',
   };
+
+  // There are anywhere between 1 to 128 quantiles in any given video, 128 is max for every video
+  const MAX_DURATION_IN_QUANTILES = 128;
   // query params instance.
   let uri;
-  // Current time for live streams.
-  let current;
+  // Time watched after last t event was sent
+  let timeWatched = 0;
+  // Last progress of the video stored
+  let lastVp = 0;
+  // Next quantile to send t event
+  let nextQuantile;
+  // Seeking state, gets reset after 1 sec of the latest seeked event
+  let isSeeking = false;
+  // Interval for Live Streams seeking state
+  let liveInterval = null;
+  // Timeout for seeking state
+  let seekingTimeout = null;
+
+  // Here we convert seconds watched to quantiles, the units accepted by the analytics service (res is 0 - 128)
+  function getCurrentProgressQuantile(progress, duration) {
+    return Math.floor(MAX_DURATION_IN_QUANTILES * (progress / duration));
+  }
+
+  // We convert `q` metric to quantiles (0 - 128) to define next breakpoint for `t` event
+  function getNextTriggerQuantile(progress, duration) {
+    return (Math.ceil((progress / duration) * uri.q) * MAX_DURATION_IN_QUANTILES) / uri.q;
+  }
+
+  // Sends the last ping when closing the player or window
+  function sendRemainingData() {
+    if (uri.pw === -1) {
+      clearLiveInterval();
+    } else {
+      const pw = getCurrentProgressQuantile(lastVp, uri.vd);
+      uri.pw = pw;
+    }
+
+    clearSeekingTimeout();
+
+    if (timeWatched) {
+      uri.ti = Math.floor(timeWatched);
+      timeWatched = 0;
+
+      sendData('t');
+    }
+  }
+
+  // We set interval for sending t events for live streams where we can't get progress info
+  function setLiveInterval() {
+    liveInterval = setInterval(() => {
+      timeWatched += 1;
+    }, 1000);
+  }
+
+  // We clear interval after complete or when seeking
+  function clearLiveInterval() {
+    clearInterval(liveInterval);
+    liveInterval = null;
+  }
+
+  // There is currently a 1 sec debounce surrounding seeking event in order to logically group multiple `seeked` events
+  function setSeekingTimeout() {
+    seekingTimeout = setTimeout(() => {
+      isSeeking = false;
+      // Set new timeout when seeked event reached for live events
+      if (uri.pw === -1 && !liveInterval) {
+        setLiveInterval();
+      }
+      sendData('vs');
+    }, 1000);
+  }
+
+  // We clear timeout after complete event or when new seeking events are received
+  function clearSeekingTimeout() {
+    clearTimeout(liveInterval);
```

Comment:
```suggestion
    clearTimeout(seekingTimeout);
```

Reply:
Good catch, thank you

Reply:
Actually github caught it. It was flagged as an assigned but unused variable. :-)